app/tests/hanging_protocols_tests/test_forms.py

```diff
@@ -294,3 +311,358 @@ def test_hanging_protocol_clientside():
         }
     )
     assert form.is_valid()
+
+
+def make_ci_list(
+    *,
```

Comment:
I used keyworded arguments here to make sure the number of ci by type is correctly passed, to avoid mistakes since the four arguments are integers.